cmd/frontend/internal/frontend/metrics.go

```diff
@@ -0,0 +1,260 @@
+		meridioMetrics.MERIDIO_ATTRACTOR_GATEWAY_ROUTES_EXPORTED,
+		metric.WithUnit("routes"),
+		metric.WithDescription("Counts number of routes exported for a gateway."),
+		metric.WithInt64Callback(func(ctx context.Context, observer metric.Int64Observer) error {
+			return gm.observe(
+				ctx,
+				observer,
+				func(birdStats *BirdStats) int64 {
+					return int64(birdStats.routesExported)
+				},
+				lp,
+			)
+		}),
+	)
+	if err != nil {
+		return fmt.Errorf("frontend metrics, failed to Int64ObservableCounter (%s): %w", meridioMetrics.MERIDIO_ATTRACTOR_GATEWAY_ROUTES_EXPORTED, err)
+	}
+
+	_, err = gm.meter.Int64ObservableCounter(
+		meridioMetrics.MERIDIO_ATTRACTOR_GATEWAY_ROUTES_PREFERRED,
+		metric.WithUnit("routes"),
+		metric.WithDescription("Counts number of routes preferred for a gateway."),
+		metric.WithInt64Callback(func(ctx context.Context, observer metric.Int64Observer) error {
+			return gm.observe(
+				ctx,
+				observer,
+				func(birdStats *BirdStats) int64 {
+					return int64(birdStats.routesPreferred)
+				},
+				lp,
+			)
+		}),
+	)
+	if err != nil {
+		return fmt.Errorf("frontend metrics, failed to Int64ObservableCounter (%s): %w", meridioMetrics.MERIDIO_ATTRACTOR_GATEWAY_ROUTES_PREFERRED, err)
+	}
+	return nil
+}
+
+func (gm *GatewayMetrics) observe(ctx context.Context, observer metric.Int64Observer, valueFunc func(*BirdStats) int64, lp string) error {
+	gm.mu.Lock()
+	defer gm.mu.Unlock()
+
+	spaOutput, err := gm.RoutingService.ShowProtocolSessions(ctx, lp, "")
+	if err != nil {
+		return fmt.Errorf("frontend metrics, failed to RoutingService.ShowProtocolSessions: %w", err)
+	}
+
+	birdStats := ParseShowProtocolsAll(spaOutput)
+
+	for _, birdStat := range birdStats {
+		gateway, exists := gm.gateways[birdStat.gatewayName]
+		if !exists {
+			continue
+		}
+
+		metricAttributes := []metric.ObserveOption{
+			metric.WithAttributes(attribute.String("Protocol", getProtocolName(gateway))),
+			metric.WithAttributes(attribute.String("Gateway", gateway.Name)),
+			metric.WithAttributes(attribute.String("IP", gateway.Address)),
+		}
+		metricAttributes = append(metricAttributes, gm.metricAttributes...)
+
+		observer.Observe(
+			valueFunc(birdStat),
+			metricAttributes...,
+		)
+	}
+
+	return nil
+}
+
+func getProtocolName(gateway *nspAPI.Gateway) string {
+	res := "unknown"
+	switch strings.ToLower(gateway.Protocol) {
+	case "bgp":
+		res = "BGP"
+	case "static":
+		res = "Static"
+	}
+	if gateway.Bfd {
+		res = fmt.Sprintf("%s+BFD", res)
+	}
+	return res
+}
+
+type BirdStats struct {
+	gatewayName     string
+	routesImported  int
+	routesExported  int
+	routesPreferred int
+}
+
+func (bs *BirdStats) String() string {
+	return fmt.Sprintf("%s (%d, %d, %d)", bs.gatewayName, bs.routesImported, bs.routesExported, bs.routesPreferred)
+}
+
+func ParseShowProtocolsAll(output string) []*BirdStats {
+	res := []*BirdStats{}
+
+	protocols := strings.Split(output, "\n\n")
+
+	regex := regexp.MustCompile(`(?m)    Routes:.*`)
```

Comment:
Because the regex does not change, performance wise it better to create a file local variable instead of compiling it on every run.